deps/compiler-rt.mk

```diff
@@ -0,0 +1,144 @@
+##
+# In order to support fallbacks within llvm we need to support
+# compiler-rt. This means linking sys.so against it and resolving
+# symbols at during JIT. For the latter part we need to create
+# a .so that we can load, but compiler-rt only comes in a .a.
+#
+# There are several configurations to take into account.
+# 1. USE_SYSTEM_LLVM == 1
+#    libclang_rt.builtins is distributed with clang and so
+#    we assume that USE_SYSTEM_LLVM == 1 means that clang is also
+#    installed.
+# 2. BUILD_LLVM_CLANG == 1
+#    libclang_rt.builtins is build with clang and we can pick that up.
+# 3. BUILD_COMPILER_RT == 1
+#    Download and install compiler_rt independently of llvm/clang
+#
+# Since we need the shared opjectfile for JIT there is no USE_SYSTEM_COMPILER_RT
```

Comment:
worth a shot, I'd rather carry patches and maybe adjust the way we build LLVM if the long-term idea is we can upstream the patches, instead of needing extra makefile wrappers here